scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/CollectionConversionExt.scala

```diff
@@ -0,0 +1,118 @@
+package pl.touk.nussknacker.engine.extension
+
+import cats.data.ValidatedNel
+import cats.implicits.catsSyntaxValidatedId
+import pl.touk.nussknacker.engine.api.generics.{GenericFunctionTypingError, MethodTypeInfo}
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedObjectTypingResult, TypingResult, Unknown}
+import pl.touk.nussknacker.engine.definition.clazz.{ClassDefinitionSet, FunctionalMethodDefinition, MethodDefinition}
+import pl.touk.nussknacker.engine.extension.ToMapConversion.{keyName, valueName}
+import pl.touk.nussknacker.engine.util.classes.Extensions.ClassExtensions
+
+import java.util.{Collection => JCollection, List => JList, Map => JMap}
+
+class CollectionConversionExt(target: Any) {
+
+  def isList(): Boolean         = ToListConversion.canConvert(target)
+  def toList(): JList[_]        = ToListConversion.convert(target)
+  def toListOrNull(): JList[_]  = ToListConversion.convertOrNull(target)
+  def isMap(): Boolean          = ToMapConversion.canConvert(target)
+  def toMap(): JMap[_, _]       = ToMapConversion.convert(target)
+  def toMapOrNull(): JMap[_, _] = ToMapConversion.convertOrNull(target)
+
+}
+
+object CollectionConversionExt extends ExtensionMethodsHandler {
+  private val unknownClass    = classOf[Object]
+  private val booleanTyping   = Typed.typedClass[Boolean]
+  private val collectionClass = classOf[JCollection[_]]
+
+  private val isMapMethodDefinition = FunctionalMethodDefinition(
+    typeFunction = (invocationTarget, _) => toMapTypeFunction(invocationTarget).map(_ => booleanTyping),
+    signature = MethodTypeInfo.noArgTypeInfo,
```

Comment:
We should return the correct type in the signature because it is shown in suggestions mechanism. Here should be boolean, below should be `JMap[Unkonwn, Unkonwn]` etc.

Reply:
Done